spec/models/product_importer_spec.rb

```diff
@@ -317,6 +319,33 @@
     end
   end
 
+  describe "updating non-updatable fields on existing products" do
+    before do
+      csv_data = CSV.generate do |csv|
+        csv << ["name", "supplier", "category", "on_hand", "price", "units", "unit_type"]
+        csv << ["Beetroot", "And Another Enterprise", "Meat", "5", "3.50", "500", "g"]
+        csv << ["Tomato", "And Another Enterprise", "Vegetables", "6", "5.50", "500", "Kg"]
+      end
+      File.write('/tmp/test-m.csv', csv_data)
+      file = File.new('/tmp/test-m.csv')
```

Comment:
maiking use of Rails' tmp/ folder seems more appropriate. If something happens and you spot this file in tmp/ you know it's something related to the app.

Reply:
Product import uses the rails tmp folder for it's destination of uploaded files, but this is purely for the specs to create a CSV file first on the disk, and then "upload" or use it during the test. I thought the system /tmp folder would be better, as it will get cleared automatically regardless of what machine the specs are running on.